language-support/scala/codegen/codegen.bzl

```
@@ -0,0 +1,89 @@
+# Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+def _dar_to_scala_impl(ctx):
+    codegen_out_dir = ctx.outputs.codegen_out
+    srcjar_out_dir = ctx.outputs.srcjar_out
```

```suggestion
    srcjar_out_file = ctx.outputs.srcjar_out
```
Misleading name. It's the output `srcjar` file, not a directory.